File: old/src/queue/queue_menu.cpp
#include <iostream>

#include "queue_menu.hpp"
#include "queue.hpp"

using namespace std;

void QueueMenu::show() {
    int option;
    
    Queue queue1;
    QueueItem item;

    cout << "Queue generator\n";

    do {
        cout << "1 - Enqueue element\n";
        cout << "2 - Dequeue element\n";
        cout << "3 - Print queue\n";
        cout << "\n0 - Stop program\n";
        cout << "Option: ";

        cin >> option;

        if (option == 1) {
            cout << "\nElement to enqueue: ";
            cin >> item;
            queue1.enqueue(item);
        } else if (option == 2) {
            item = queue1.dequeue();
            cout << "\nRemoved element: " << item << endl;
            queue1.print();
        } else if (option == 3) {
            queue1.print();
        } else if (option == 0) {
            exit(0);
        } else {
            cout << "\nInvalid option!\n";
        }
    } while (option != 0);
}

File: src/queue/queue_menu.rs
use std::io;

use super::queue::Queue;

pub fn start_queue() {
    let mut queue: Queue = Queue::new();

    fn load_argument() -> i32 {
        let mut option: String = String::new();

        io::stdin()
            .read_line(&mut option)
            .expect("Failed to read line");
        option.trim().parse().expect("Please type a number!")
    }

    println!("Queue generator");

    loop {
        println!("1 - Enqueue element");
        println!("2 - Dequeue element");
        println!("3 - Print queue");
        println!("0 - Stop program");

        let option: i32 = load_argument();

        match option {
            1 => {
                println!("Element to enqueue: ");
                let element: i32 = load_argument();
                queue.enqueue(element);
                continue;
            }
            2 => {
                println!("Dequeued element: {}", queue.dequeue());
                continue;
            }
            3 => {
                queue.print();
                continue;
            }
            0 => {
                println!("Program stopped!");
                return;
            }
            _ => {
                println!("Invalid option!");
                continue;
            }
        }
    }
}
